src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -8188,8 +8188,8 @@ public static VarHandle permuteCoordinates(VarHandle target, List<Class<?>> newC
      * filter function and the target var handle is then called on the modified (usually shortened)
      * coordinate list.
      * <p>
-     * If {@code R} is the return type of the filter (which cannot be void), the target var handle must accept a value of
-     * type {@code R} as its coordinate in position {@code pos}, preceded and/or followed by
+     * If {@code R} is the return type of the filter, then, unless {@code R} is void, the target var handle must accept
```

Comment:
Not introduced by you - but this para seems to be repeated twice-ish. I think we should try to consolidate the two, if we can.